fog/test-client/src/bin/main.rs

```diff
@@ -2,38 +2,153 @@
 
 //! Test Client
 
-use mc_common::logger::{create_app_logger, o};
-
-use mc_fog_test_client::{config::Config, error::TestClientError, test_client::TestClient};
+use mc_common::logger::{create_app_logger, log, o};
 
+use core::convert::TryFrom;
+use grpcio::{RpcStatus, RpcStatusCode};
+use mc_fog_test_client::{
+    config::TestClientConfig,
+    error::TestClientError,
+    test_client::{TestClient, TestClientPolicy},
+};
+use mc_sgx_css::Signature;
+use mc_util_grpc::AdminServer;
+use serde::Serialize;
+use std::sync::Arc;
 use structopt::StructOpt;
 
+#[derive(Serialize, Debug, Clone)]
+struct JsonData {
+    pub policy: TestClientPolicy,
+    pub config: TestClientConfig,
+}
+
 fn main() {
     mc_common::setup_panic_handler();
     let (logger, _global_logger_guard) = create_app_logger(o!());
 
-    let config = Config::from_args();
+    let config = TestClientConfig::from_args();
+
+    // Set up test client policy taking into account the runtime config values
+    let policy = TestClientPolicy {
+        // Don't fail fast when running continuously, we want to keep measuring after the deadline
+        fail_fast_on_deadline: !config.continuous,
+        // Don't test RTH memos when passed --no_memos
+        test_rth_memos: !config.no_memos,
+        tx_submit_deadline: config.consensus_wait,
+        tx_receive_deadline: config.consensus_wait,
+        double_spend_wait: config.ledger_sync_wait,
+        transfer_amount: config.transfer_amount,
+        ..Default::default()
+    };
 
     let account_keys = config.load_accounts(&logger);
 
-    match TestClient::new(
+    // Load any css from disk
+    let consensus_sigstruct = config
+        .consensus_enclave_css
+        .as_ref()
+        .map(load_css_file)
+        .transpose()
+        .expect("loading css failed");
+    let fog_ingest_sigstruct = config
+        .fog_ingest_enclave_css
+        .as_ref()
+        .map(load_css_file)
+        .transpose()
+        .expect("loading css failed");
+    let fog_ledger_sigstruct = config
+        .fog_ledger_enclave_css
+        .as_ref()
+        .map(load_css_file)
+        .transpose()
+        .expect("loading css failed");
+    let fog_view_sigstruct = config
+        .fog_view_enclave_css
+        .as_ref()
+        .map(load_css_file)
+        .transpose()
+        .expect("loading css failed");
```

Comment:
Optional: This "main" method is 120 lines, which is on the  longer side. Since this block of code is self-contained and requires a comment currently, consider extracting to a helper method with a meaningful name. This would increase readability IMO and remove the need for a comment.